src/main/java/tc/oc/pgm/match/MatchImpl.java

```diff
@@ -809,6 +797,23 @@ public void unload() {
     world.enqueue();
 
     loaded.compareAndSet(true, false);
+  }
+
+  @Override
+  public void destroy() {
+    if (isLoaded())
+      throw new IllegalStateException("Can't destroy a match that has not been unloaded");
```

Comment:
How about it unloads if it isn't already. I don't think destroy should ever fail, similarly with unload.

Reply:
That seems fine, but since destroy is called asyncronously, it may cause other issues. I'll log it as severe and call unload